klarna-on-demand/src/main/java/com/klarna/ondemand/OriginProof.java

```diff
@@ -0,0 +1,64 @@
+package com.klarna.ondemand;
+
+import android.util.Base64;
+import org.json.JSONObject;
+import java.util.UUID;
+
+/**
+ * Assists in creating origin proofs that allow performing secure purchases.
```

Comment:
This doesn't so much assist, as actually serve as an embodiment of what you describe here.
These are actual objects that serve as an origin proof.